### PR TITLE
Update plugin interfaces

### PR DESCRIPTION
**Pull request by yruslan**
* Interfaces are updated to align with Enceladus
* IntelliJ IDEA project files are removed from the repository. The POM file can be used to open the project in IntelliJ IDEA.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,11 +1,51 @@
...
+# other scm
+.svn
+.CVS
+.hg*
+
+# switch to regexp syntax.
+#  syntax: regexp
+#  ^\.pc/
+
+#SHITTY output not in target directory
```
Comment (Zejnilovic) [inline .gitignore]:
```suggestion
output not in target directory
```

I blame @felipemmelo

Comment (yruslan) [reply .gitignore]: 😆  Omg, Thanks!

Comment (TarannumPraveen) [reply .gitignore]: Sorry my mistake

Comment (yruslan) [reply .gitignore]: No worries

**Pull request merged**